### PR TITLE
Adds the ability to set UID/GID used for connection to the NFS server

### PR DESCRIPTION
**Pull request by pduveau**
## Description
Adds the possibility to set the UID/GID used for the connection to the NFS server. These two new parameters are added to the NFS client service: the UID value and the GID value. They are conditioned by a boolean parameter which if disabled leaves the UID/GID defined by the underlying OS.

## Motivation and context
Les UID/GID attribués par Android ne sont pas nécessairement identiques d'une installation à une autre. Ainsi lorsque les librairies et base de données sont partagées entre plusieurs installation de Kodi sur différents OS, il devient complexe voire impossible de même en place des ACLs fonctionnels pour toutes les différentes installations sauf à tout surcharger avec les fonctionnalités de "squash" du serveur NAS. Avec cette capacité, il est désormais possible que toutes les installations utilisent les mêmes UID/GID et ainsi simplfie les ACLs 

## How has this been tested?
The test was done using the settings file widget to create a folder on the NAS (without squash option in the export) and then check that the UID/GID of the folder on the NAS file system are identical with those defined.

## What is the effect on users?
The effect is a simpler procedure to set up a shared library between different kodi's installations.

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -22069,7 +22069,43 @@ msgctxt "#37052"
...
+#. Description of setting #37055 NFS connection User ID (UID)"
+#: xbmc/filesystem/NFSFile.cpp
+msgctxt "#37056"
+msgid "NFS User ID (UID) to be used by NFS connections. This UID must match a UID of the NAS system (/etc/passwd). The value is between 1000 and 60000."
+msgstr ""
+
+#. Setting #37055 NFS connection Group ID (GID)"
+#: xbmc/filesystem/NFSFile.cpp
+msgctxt "#37057"
+msgid "NFS connection Goup ID (GID)"
```
Comment (lrusak) [inline addons/resource.language.en_gb/resources/strings.po:22099]:
```suggestion
msgid "NFS connection Group ID (GID)"
```

File: addons/resource.language.en_gb/resources/strings.po [hunk 2]
```diff
@@ -22069,7 +22069,43 @@ msgctxt "#37052"
...
+msgstr ""
+
+#. Description of setting #37057 NFS connection Group ID (GID)"
+#: xbmc/filesystem/NFSFile.cpp
+msgctxt "#37058"
+msgid "NFS Goup ID (GID) to be used by NFS connections. This GID must match a GID of the NAS system (/etc/group). The value is between 100 and 60000."
```
Comment (lrusak) [inline addons/resource.language.en_gb/resources/strings.po:22105]:
```suggestion
msgid "NFS Group ID (GID) to be used by NFS connections. This GID must match a GID already present on the NFS server. The value is between 100 and 60000."
```

File: addons/resource.language.en_gb/resources/strings.po [hunk 3]
```diff
@@ -22069,7 +22069,43 @@ msgctxt "#37052"
...
+#. Setting #37055 NFS connection User ID (UID)"
+#: xbmc/filesystem/NFSFile.cpp
+msgctxt "#37055"
+msgid "NFS connection User ID (UID)"
+msgstr ""
+
+#. Description of setting #37055 NFS connection User ID (UID)"
+#: xbmc/filesystem/NFSFile.cpp
+msgctxt "#37056"
+msgid "NFS User ID (UID) to be used by NFS connections. This UID must match a UID of the NAS system (/etc/passwd). The value is between 1000 and 60000."
```
Comment (lrusak) [inline addons/resource.language.en_gb/resources/strings.po:22093]:
```suggestion
msgid "NFS User ID (UID) to be used by NFS connections. This UID must match a UID already present on the NFS server. The value is between 1000 and 60000."
```

File: xbmc/filesystem/NFSFile.cpp [hunk 4]
```diff
@@ -553,6 +556,19 @@ void CNfsConnection::setOptions(struct nfs_context* context)
   }
 
   CLog::Log(LOGDEBUG, "NFS: version: {}", nfsVersion);
+
+  const bool isNfsEnableIDs = settings->GetBool(SETTING_NFS_ENABLE_IDS);
```
Comment (lrusak) [inline xbmc/filesystem/NFSFile.cpp:560]:
```suggestion
  const bool isNfsIdsEnabled = settings->GetBool(SETTING_NFS_ENABLE_IDS);
```

Comment (lrusak) [inline xbmc/filesystem/NFSFile.cpp:560]:
```suggestion
  const bool isNfsIdsEnabled = settings->GetBool(SETTING_NFS_ENABLE_IDS);
```

**Review by lrusak — COMMENTED** [review]
Looks great! Thanks for the contribution! I left a few comments about wording. The functional changes look good though.

Please also apply the clang-format diff.

File: system/settings/settings.xml [hunk 5]
```diff
@@ -2383,6 +2383,37 @@
...
+          <default>false</default>
+          <control type="toggle" />
+        </setting>
+        <setting id="nfs.uid" type="integer" label="37055" help="37056" parent="nfs.enableids">
+          <level>3</level>
+          <default>1000</default>
+          <constraints>
+            <minimum>1000</minimum>
+            <step>1</step>
+            <maximum>60000</maximum>
```
Comment (wsnipex) [inline system/settings/settings.xml:2397]: although this is the default max gid value, this can be adjusted, so we shouldn't enforce a hard limit here.

**Comment by pduveau** [comment]
Something is not running well in my configuration. So I am not sure that my proposal is handling all use cases. I have more to debug first. I will come back as soon as possible with a fully functionnal enhancement.
Thank you for your reviews 
Best regards
Philippe

**Comment by pduveau** [comment]
I found a part of the bug in the V4 implementation, the file are created on remote server with a null permission 0000 so when kodi try to write in it gets a permission denied. In V3, it work ! But the permissions are not exactly what I am looking for. The bug and convenience are not linked to my proposal but I'll try to solve those points first.

**Pull request closed without merging**